### PR TITLE
Default enable log recycling

### LINKED ISSUE #263: Default enable log recycling.
## Description
According to PR: #224 , we have make RaftEngine support log recycling with configuration `enable-log-recycle: true`. Also, the testing results shown in this pr have proved that this feature have positive effects to `write` workloads.
So, starting from `0.3` version, we want to make it `open` as default.

## Related changes
Default values to `format-version` and `enable-log-recycle` should be modified respectively:
* Default value of `format-version` should be changed to `2` (origin is `1`);
* Default value of `enable-log-recycle` should be changed to `true` (origin is `false`);

### PR DESCRIPTION
**Pull request by LykxSassinator**
## Description
Default enable log recycling. Close #263.

Signed-off-by: Lucasliang <nkcs_lykx@hotmail.com>

### CODE REVIEW

File: tests/failpoints/test_engine.rs [hunk 1]
```diff
@@ -611,6 +611,7 @@ fn test_recycle_with_stale_logbatch_at_tail() {
         target_file_size: ReadableSize::kb(2),
         purge_threshold: ReadableSize::kb(4),
         enable_log_recycle: true,
+        format_version: Version::V1,
```
Comment (tabokie) [inline tests/failpoints/test_engine.rs:614]: Why?

Comment (LykxSassinator) [reply tests/failpoints/test_engine.rs:614]: Here, default Version has been updated to V2 to enable log recycling defaultly, this case should be compatible to this modification.

The original design of this case is to mock a case that RaftEngine does recycling on `V1` log files, and it cannot recognize the recycled part in the recycled file, as `V1` logs has no signatures.

**Review by tabokie — APPROVED** [review]

**Review by tabokie — COMMENTED** [review]
Update the changelog under "Behavior changes".

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## [Unreleased]
 
+### Behavior Changes
+* Enable log recycling by default.
```
Comment (tabokie) [inline CHANGELOG.md:6]: And put it at the end.
```suggestion
### Behavior Changes

* Enable log recycling by default.
```

Comment (tabokie) [reply CHANGELOG.md]: @LykxSassinator I updated the comment.

Comment (LykxSassinator) [reply CHANGELOG.md]: How about => 

### Bahavior Changes

* Change format version to 2 from 1 by default.
* Enable log recycling by default.

Comment (tabokie) [reply CHANGELOG.md]: Sure.

**Review by tabokie — APPROVED** [review]

**Pull request merged**